Makefile

```diff
@@ -1,7 +1,35 @@
 SHELL := /bin/bash -o pipefail -o errexit
 
+# You can override all of these variables on the command line like so:
+# ENV_NAME=dev TMPDIR=$HOME make test
+ENV_NAME ?= "conda-build"
+DOC_ENV_NAME ?= "conda-build-docs"
+TMPDIR ?= /tmp
+PYTHON_VERSION ?= 3.8
+
+# Setup env for documents
 env-docs:
-	conda create --name conda-build-docs --channel defaults python=3.8 --yes
-	conda run --name conda-build-docs pip install -r ./docs/requirements.txt
+	conda create --name $(DOC_ENV_NAME) --channel defaults python=$(PYTHON_VERSION) --yes
+	conda run --name $(DOC_ENV_NAME) pip install -r ./docs/requirements.txt
 
 .PHONY: $(MAKECMDGOALS)
+
+# Runs all tests
+.PHONY: test
+test: ../conda_build_test_recipe
+	conda run --live-stream -n $(ENV_NAME) pytest tests/ --basetemp $(TMPDIR)
+
+# Run the serial tests
+.PHONY: test-serial
+test-serial: ../conda_build_test_recipe
+	conda run --live-stream -n $(ENV_NAME) pytest tests/ -m "serial" --basetemp $(TMPDIR)
+
+# Run the not serial tests
+.PHONY: test-not-serial
+test-not-serial: ../conda_build_test_recipe
+	conda run --live-stream -n $(ENV_NAME) pytest tests/ -m "not serial" --basetemp $(TMPDIR)
```

Comment:
When I was in my `conda-build` env that I use for testing, all of the `make` targets would fail with: 
```
EnvironmentLocationNotFound: Not a conda environment: /Users/biancahenderson/miniconda/envs/conda-build/envs/conda-build
```
Looks like it's unable to detect if the activated env is `conda-build`, and I'm not sure how to fix this without changing the target commands.

Reply:
Interesting. I wonder if that is a conda run bug because it works for me when I'm in my `conda-build` env.